### DISCUSSION TITLE
Two factors and external logins

### DISCUSSION THREAD

**Discussion by rafik4000** [HBSTech/Kentico13CoreBaseline#8] [Q&A]
Hey, I have a question is it possbile to integrate Two factors and external login in the project?

**Comment by KenticoDevTrev** [comment]
Ide love to, do you have working code we can use?

**Reply by rafik4000** [reply]
I saw the work done on medioclinic on https://github.com/Kentico/xperience-training-13

**Reply by KenticoDevTrev** [reply]
Our new developer who just did the training is working on integrating the external logins, hopefully we'll have something within a couple weeks as part of the baseline!

**Reply by KenticoDevTrev** [reply]
Just an update, one of our team members was able to get the various External Authentication methods going (Google, Facebook, Microsoft, and Twitter), we'll be adding it to the baseline with instructions hopefully by Tuesday.  We were not able to get the 2 factor auth with the normal Kentico login set up however, however using an external authenticator should be a good solution for those.  

We may add in Email based 2 form authentication as it doesn't look terribly hard...but SMS two form is tricky.
